crates/nix_rs/src/env.rs

```diff
@@ -196,6 +202,86 @@ impl OS {
     }
 }
 
+/// The installer used to install Nix (applicable only for non-NixOS systems)
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum NixInstaller {
+    /// The installer from <https://github.com/DeterminateSystems/nix-installer>
+    DetSys { version: NixInstallerVersion },
+    /// Either offical installer or from a different package manager
+    Other,
+}
+
+impl Display for NixInstaller {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NixInstaller::DetSys { version } => {
+                write!(f, "nix-installer ({})", version)
+            }
+            NixInstaller::Other => write!(f, "Unknown installer"),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, PartialOrd, Eq, Clone)]
+pub struct NixInstallerVersion {
```

Comment:
```suggestion
pub struct DetSysNixInstallerVersion {
```

Reply:
Could you move DetSys specific code out of this file into its own module (`detsys_installer.rs`)?

Reply:
Later, outside this PR, I'll make `env.rs` a directory module (`env/*`).

Reply:
https://github.com/juspay/omnix/pull/213/commits/0e8a14d6b4ffada6b6391ac5123ad36bf0e81dd8

Reply:
https://github.com/juspay/omnix/pull/213/commits/d006fc4d7d5c01b4698c2a11057bd26f2b996f5b

Reply:
It has not been _completely_ isolated/ encapsulated away though; `env.rs` still [complects](https://www.infoq.com/presentations/Simple-Made-Easy/) with det sys stuff. I'll refactor and push; you can then review those commits.